wagtail/contrib/modeladmin/views.py

```diff
@@ -859,39 +856,37 @@ def get_field_display_value(self, field_name, field=None):
                 return val_funct()
             return val_funct
 
-        # If we have a real field, we can utilise that to try to display
-        # something more useful
-        if field is not None:
-            try:
-                field_type = field.get_internal_type()
-                if (
-                    field_type == 'ForeignKey' and
-                    field.related_model == get_image_model()
-                ):
-                    # The field is an image
-                    return self.get_image_field_display(field_name, field)
-
-                if (
-                    field_type == 'ForeignKey' and
-                    field.related_model == get_document_model()
-                ):
-                    # The field is a document
-                    return self.get_document_field_display(field_name, field)
-
-            except AttributeError:
-                pass
+        # Now let's get the attribute value from the instance itself and see if
+        # we can render something useful. raises AttributeError appropriately.
+        val = getattr(self.instance, field_name)
+
+        # wagtail.wagtailimages might not be installed
+        try:
+            from wagtail.wagtailimages.models import AbstractImage
```

Comment:
Doesn't this give a ``RuntimeError`` since wagtailimages is installed within the wagtail package, so it can be imported but can't load the model since it's not installed in django.

```console
>>> try:
...     from wagtail.wagtailimages.models import AbstractImage
... except ImportError:
...     print('import error happened')
...
Traceback (most recent call last):
  File "<console>", line 2, in <module>
  File "/Users/m.dingjan/.pyenv/versions/3.5.3/lib/python3.5/site-packages/wagtail/wagtailimages/models.py", line 337, in <module>
    class Image(AbstractImage):
  File "~/.pyenv/versions/3.5.3/lib/python3.5/site-packages/django/db/models/base.py", line 113, in __new__
    "INSTALLED_APPS." % (module, name)
RuntimeError: Model class wagtail.wagtailimages.models.Image doesn't declare an explicit app_label and isn't in an application in INSTALLED_APPS.```